beacon-chain/sync/fork_watcher.go

```diff
@@ -38,101 +41,88 @@ func (s *Service) forkWatcher() {
 }
 
 // registerForUpcomingFork registers appropriate gossip and RPC topic if there is a fork in the next epoch.
```

Comment:
The name and the godoc of this functions do not describe the function:
Name and godoc talk about the next fork, while the function takes care of the current AND the next fork.